OpenRA.Platforms.Default/Sdl2PlatformWindow.cs

```diff
@@ -120,6 +118,25 @@ public Sdl2PlatformWindow(Size requestWindowSize, WindowMode windowMode, int bat
 					windowSize = new Size(display.w, display.h);
 				}
 
+				// Windows and Linux define window sizes in native pixel units.
+				// Query the display/dpi scale so we can convert our requested effective size.
+				// This is not necessary on macOS, which defines window sizes in effective units ("points").
+				if (Platform.CurrentPlatform == PlatformType.Windows)
+				{
+					float ddpi, hdpi, vdpi;
+					if (SDL.SDL_GetDisplayDPI(0, out ddpi, out hdpi, out vdpi) == 0)
+						windowScale = ddpi / 96;
+				}
+				else if (Platform.CurrentPlatform != PlatformType.OSX)
+				{
+					float scale = 1;
+					var scaleVariable = Environment.GetEnvironmentVariable("OPENRA_DISPLAY_SCALE");
+					if (scaleVariable != null && float.TryParse(scaleVariable, out scale))
```

Comment:
This probably needs an invariant culture as testing on German locale laptop crashes unexpectedly:

```
OPENRA_DISPLAY_SCALE=1.5 ./launch-game.sh 
Platform is Linux
Engine version is 6497d1ed5d81e6be209e464e007f35d0773b1877
Using SDL 2 with OpenGL renderer
Desktop resolution: 1920x1080
No custom resolution provided, using desktop resolution
Using resolution: 20490x11520
Using window scale 15,00
Exception of type `System.InvalidOperationException`: Can not create OpenGL context. (Error: Invalid window)
  at OpenRA.Platforms.Default.Sdl2GraphicsContext.InitializeOpenGL () [0x000c1] in /home/matthias/Entwicklung/OpenHV/engine/OpenRA.Platforms.Default/Sdl2GraphicsContext.cs:53 
  at OpenRA.Platforms.Default.ThreadedGraphicsContext.RenderThread (System.Object contextObject) [0x0002a] in /home/matthias/Entwicklung/OpenHV/engine/OpenRA.Platforms.Default/ThreadedGraphicsContext.cs:79 
  at System.Threading.ThreadHelper.ThreadStart_Context (System.Object state) [0x00025] in /home/abuild/rpmbuild/BUILD/mono-6.4.0.198/mcs/class/referencesource/mscorlib/system/threading/thread.cs:78 
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00071] in /home/abuild/rpmbuild/BUILD/mono-6.4.0.198/mcs/class/referencesource/mscorlib/system/threading/executioncontext.cs:968 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00000] in /home/abuild/rpmbuild/BUILD/mono-6.4.0.198/mcs/class/referencesource/mscorlib/system/threading/executioncontext.cs:910 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state) [0x0002b] in /home/abuild/rpmbuild/BUILD/mono-6.4.0.198/mcs/class/referencesource/mscorlib/system/threading/executioncontext.cs:899 
  at System.Threading.ThreadHelper.ThreadStart (System.Object obj) [0x0000f] in /home/abuild/rpmbuild/BUILD/mono-6.4.0.198/mcs/class/referencesource/mscorlib/system/threading/thread.cs:93 

[ERROR] FATAL UNHANDLED EXCEPTION: System.InvalidOperationException: Can not create OpenGL context. (Error: Invalid window)
  at OpenRA.Platforms.Default.Sdl2GraphicsContext.InitializeOpenGL () [0x000c1] in /home/matthias/Entwicklung/OpenHV/engine/OpenRA.Platforms.Default/Sdl2GraphicsContext.cs:53 
  at OpenRA.Platforms.Default.ThreadedGraphicsContext.RenderThread (System.Object contextObject) [0x0002a] in /home/matthias/Entwicklung/OpenHV/engine/OpenRA.Platforms.Default/ThreadedGraphicsContext.cs:79 
  at System.Threading.ThreadHelper.ThreadStart_Context (System.Object state) [0x00025] in /home/abuild/rpmbuild/BUILD/mono-6.4.0.198/mcs/class/referencesource/mscorlib/system/threading/thread.cs:78 
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00071] in /home/abuild/rpmbuild/BUILD/mono-6.4.0.198/mcs/class/referencesource/mscorlib/system/threading/executioncontext.cs:968 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00000] in /home/abuild/rpmbuild/BUILD/mono-6.4.0.198/mcs/class/referencesource/mscorlib/system/threading/executioncontext.cs:910 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state) [0x0002b] in /home/abuild/rpmbuild/BUILD/mono-6.4.0.198/mcs/class/referencesource/mscorlib/system/threading/executioncontext.cs:899 
  at System.Threading.ThreadHelper.ThreadStart (System.Object obj) [0x0000f] in /home/abuild/rpmbuild/BUILD/mono-6.4.0.198/mcs/class/referencesource/mscorlib/system/threading/thread.cs:93
```

but works fine using:
```
matthias@linux-tuxedo:~/Entwicklung/OpenHV> OPENRA_DISPLAY_SCALE=1,5 ./launch-game.sh  
Platform is Linux
Engine version is 6497d1ed5d81e6be209e464e007f35d0773b1877
Using SDL 2 with OpenGL renderer
Desktop resolution: 1920x1080
No custom resolution provided, using desktop resolution
Using resolution: 2049x1152
Using window scale 1,50
OpenGL version: 3.2.0 NVIDIA 440.44
Using default sound device
Cannot connect to server socket err = Datei oder Verzeichnis nicht gefunden
Cannot connect to server request channel
jack server is not running or cannot be started
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
Internal mods:
        hv: OpenHV ({DEV_VERSION})
        d2k: Dune 2000 (6497d1ed5d81e6be209e464e007f35d0773b1877)
        ts: Tiberian Sun (6497d1ed5d81e6be209e464e007f35d0773b1877)
        all: All mods (6497d1ed5d81e6be209e464e007f35d0773b1877)
        cnc: Tiberian Dawn (6497d1ed5d81e6be209e464e007f35d0773b1877)
        ra: Red Alert (6497d1ed5d81e6be209e464e007f35d0773b1877)
        modcontent: Mod Content Manager (6497d1ed5d81e6be209e464e007f35d0773b1877)
External mods:
        d2k-{DEV_VERSION}: Dune 2000 ({DEV_VERSION})
        cnc-{DEV_VERSION}: Tiberian Dawn ({DEV_VERSION})
        ts-release-20191117: Tiberian Sun (release-20191117)
        ra-df61b59746978a7187e0c5f39f7d584e249805a7: Red Alert (df61b59746978a7187e0c5f39f7d584e249805a7)
        ra-{DEV_VERSION}: Red Alert ({DEV_VERSION})
        hv-{DEV_VERSION}: OpenHV ({DEV_VERSION})
Loading mod: hv
System.Net.Sockets.SocketOptionName 0x17 is not supported at IP level
[2020-01-21T21:24:31] Game started
```

Reply:
This is just moving existing code, but yes, we can fix it here.

Reply:
Actually, it makes more sense to defer this fix to the next PR, which rewrites this code again.

Reply:
Fix included in 3b3a81cecc, which is one of the PRs that depends on this one.